c7n/resources/s3.py

```diff
@@ -178,6 +179,31 @@ def bucket_client(session, b, kms=False):
     return session.client('s3', region_name=region, config=config)
 
 
+def add_bucket_tags(session_factory, buckets, tags):
+    client = local_session(session_factory).client('s3')
+    for bucket in buckets:
+        # all the tag marshalling back and forth is a bit gross :-(
+        new_tags = {t['Key']: t['Value'] for t in tags}
+        for t in bucket.get('Tags', ()):
+            if t['Key'] not in new_tags and not t['Key'].startswith('aws'):
+                new_tags[t['Key']] = t['Value']
+        tag_set = [{'Key': k, 'Value': v} for k, v in new_tags.items()]
+        client.put_bucket_tagging(
+            Bucket=bucket['Name'], Tagging={'TagSet': tag_set})
+
+
+def remove_bucket_tags(session_factory, buckets, excludedtags):
```

Comment:
can we combine these into two methods, s/modify_bucket_tags(factory, buckets, add, remove) style